docs/index-management/index-performance.md

```diff
@@ -466,15 +466,57 @@ increases. Often, such traversals can be rewritten as constrained
 traversals that can utilize a vertex-centric index to ensure acceptable
 performance at scale.
 
+### Vertex-centric indexes on implicit keys
+
+In some cases it is relevant to find an edge based on properties of the adjacent vertex.
+Let's say we want to find out whether or not Hercules has battled Cerberus.
+```groovy
+h = g.V().has('name', 'hercules').next()
+g.V(h).out('battled').has('name', 'cerberus').limit(1).hasNext()
+```
+
+A query like this can not use a vertex centric index because it filters on vertex properties rather than edge properties.
+But by restructuring the query, we can achieve exactly this.
+As both vertices are known, the vertex ids can be used to select the edge.
+In contrast to the name "Cebereus", which is a property of the adjacent vertex, the id of this vertex is already saved within the connecting edge itself.
+Although, this id is internal and therefore not explcitly visible as a property of the edge.
+This can be solved by referring to the id as an `ImplicitKey`:
+```groovy
+h = g.V().has('name', 'hercules').next()
+c = g.V().has('name', 'cerberus').next()
+g.V(h).outE('battled').has(ImplicitKey.ADJACENT_ID.name(), c.id()).limit(1).hasNext()
+```
+
+!!! note
+    If the class ImplicitKey can not be found, use the fully qualified class name `org.janusgraph.graphdb.types.system.ImplicitKey` instead.
+
+Assuming there is a global index on the 'name' property, this already improves the performance a lot, because it's not necessary to fetch every adjacent vertex anymore.
```

Comment:
I agree with you on this. The performance gain of this solution heavily depends on the degree of the vertex for which the query is run. But as with other sorts of indexes, this is a trade-off which the user has to decide on. While in some cases querying the global index can cause a performance reducing overhead, there are other cases with higher degree vertices where this practice can have a positive effect.
I can't tell exactly at which degree this will start to take effect, because it depends mostly on environment factors like the storage backend or the overall size and structure of your graph.